code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/MessagingExtension.java

```diff
@@ -468,10 +487,7 @@ void handlePushToken(final Event event) {
         }
 
         // Update the push token to the shared state
-        final HashMap<String, Object> messagingSharedState = new HashMap<>();
-        messagingSharedState.put(
-                MessagingConstants.SharedState.Messaging.PUSH_IDENTIFIER, pushToken);
-        getApi().createSharedState(messagingSharedState, event);
+        addPushTokenToSharedState(pushToken, event);
```

Comment:
You'll need to set the token in persistence here.

Reply:
the token is added to persistence prior to this method being called in InternalMessagingUtils:
```java
 // persist the push token in the messaging named collection
        final NamedCollection messagingNamedCollection = getNamedCollection();
        messagingNamedCollection.setString(
                MessagingConstants.SharedState.Messaging.PUSH_IDENTIFIER, newPushToken);
```